tests/Unit/ParallelAlgorithms/LinearSolver/Gmres/Test_ElementActions.cpp

```diff
@@ -68,13 +68,13 @@ struct ElementArray {
           tmpl::list<
               LinearSolver::gmres::detail::NormalizeInitialOperand<
                   fields_tag, DummyOptionsGroup, Preconditioned,
-                  DummyOptionsGroup>,
+                  DummyOptionsGroup, void>,
```

Comment:
It would be nice to add support for array sections to the action testing framework. That's a bit more involved though, because there's some charm-related functionality missing (e.g. can't currently instantiate `Parallel::charm_types_with_parameters` with a mock component). I think the steps for extending the action testing framework are: (1) add support for reductions, (2) add support for more low-level charm things like a `cbase`  alias etc, (3) add support for sections and section reductions. I opened https://github.com/sxs-collaboration/spectre/issues/3368 to keep track. I've been meaning to do (1), but it didn't make it up my priority list so far. The main reason is that I haven't found the linear-solver action-tests (that would benefit from reductions support) particularly useful when debugging because the actions are just chunks of the full algorithm, and the particular chunking is an implementation detail that I've had to adjust sometimes. The full-algorithm executable tests have been a lot more useful for the parallel linear solvers specifically.

To answer your question, the sectioned GMRES algorithm is tested by `Test_MultigridPreconditionedGmresAlgorithm` (added in the last commit).